components/cloud_start_trial/cloud_start_trial_btn.tsx

```diff
@@ -48,36 +54,51 @@ const CloudStartTrialButton = ({
 }: CloudStartTrialBtnProps) => {
     const {formatMessage} = useIntl();
     const dispatch = useDispatch<DispatchFunc>();
-
+    const subscription = useSelector(selectCloudSubscription);
+    const [openBusinessEmailModal, setOpenBusinessEmailModal] = useState(false);
     const [status, setLoadStatus] = useState(TrialLoadStatus.NotStarted);
 
+    const validateBusinessEmailOnLoad = async () => {
+        const isValidBusinessEmail = await validateBusinessEmail()();
+        if (!isValidBusinessEmail) {
+            setOpenBusinessEmailModal(true);
+        }
+    };
+
+    useEffect(() => {
+        validateBusinessEmailOnLoad();
+        if (subscription === undefined) {
+            dispatch(getCloudSubscription());
+        }
```

Comment:
I'd recommend using the `useGetSubscription` hook instead of dispatching the fetch and `useSelector`